### DISCUSSION TITLE
How to setup VSCode to properly debug components in storyboard?

### DISCUSSION THREAD

**Discussion by igorgnot** [storybookjs/storybook#26153] [Help]
### Summary

At the moment breakpoints in my project components inside storyboard cant bind. How to setup launch.js to make it work?

**Comment by Marklb** [comment]
I normally just use a chrome launch config, with the url of my running Storybook.

This is from one of my projects.
```json
{
  "type": "chrome",
  "request": "launch",
  "name": "Launch Chrome against localhost",
  "url": "http://localhost:6006/",
  "webRoot": "${workspaceFolder}",
  "userDataDir": "${workspaceFolder}/.vscode/.chrome-user-dir",
},
```

I haven't done this in a while, so I just checked that it still works. The only problem that I noticed is that my VSCode is adding my breakpoints to the wrong line. That could just be a problem with my project though and my not be related to anything Storybook is doing. I just haven't looked into why the breakpoint lines numbers are off for me, yet.

I don't know if there is something specific that needs to be configured, so if it matters, I am using Angular and haven't added anything extra to allow the debugger to work, that I know of.

**Comment by jubelhart-aops** [comment]
I know this is kind of an older thread, but I also had this issue and found a solution. It's quite possible that my solution is super specific to my setup, so apologies if it doesn't work. I'm on `@storybook/nextjs` `v8.0.4` and running on `localhost` on an M1 Mac. I used the Firefox debugger extension for VS Code. https://marketplace.visualstudio.com/items?itemName=firefox-devtools.vscode-firefox-debug

I put the following config in my launch.json:
```json
{
  "type": "firefox",
  "request": "launch",
  "name": "Storybook Debug Firefox",
  "url": "http://localhost:6006", // Your Storybook URL here
  "webRoot": "${workspaceRoot}",
  "suggestPathMappingWizard": true,
}
```

I did the following steps to get it to work properly:
1. Run storybook
2. Run the VS Code debug script above from `Run and Debug` (`cmd` + `shift` + `d`). This launches Firefox in debug mode
3. Open storybook in the new Firefox window to a component I wanted to debug
4. Open the component's source file in VS Code
5. Now here's the magic that made it work. Run `Firefox: Run the path mapping wizard` in VS Code with `cmd` + `shift` + `p`.
6. This magically just figures out my webpack paths and now setting debug breakpoints works! 

This is a bit convoluted but it worked in the end! I spent another couple hours attempting to figure out Chrome, but even with the same exact path mappings I just couldn't figure it out 😅. Hopefully this helps someone else!
